### PR TITLE
Use upstream URL to find repo name

### PR DESCRIPTION
**Pull request by harterrt**

### CODE REVIEW

**Comment by harterrt** [comment]
Use the `origin` url to find the repo name instead of the current directory. For example, I cloned mozilla-reports using: `git clone mozilla/mozilla-reports rtmo`. Currently the tool would interpret the repo name as `rtmo` instead of `mozilla-reports`

**Comment by harterrt** [comment]
@jklukas - I don't seem to be able to mark you as a reviewer. FYI.

File: git-push-fork-to-upstream-branch [hunk 1]
```diff
@@ -15,10 +15,10 @@ fi
 
 SOURCE_GH_USER=$(echo "$BRANCH_SPEC" | awk -F: '{print $1}')
 SOURCE_BRANCH=$(echo "$BRANCH_SPEC" | awk -F: '{print $2}')
-REPO_NAME=$(basename $(git rev-parse --show-toplevel))
+REPO_NAME=$(git remote get-url --push origin | cut -d '/' -f 2 | cut -d '.' -f 1)
```
Comment (jklukas) [inline git-push-fork-to-upstream-branch:18]: This assumes that there's a remote called `origin`, yes? That's _probably_ always the case? I suppose we could simply document that as an assumption.

File: git-push-fork-to-upstream-branch [hunk 2]
```diff
@@ -15,10 +15,10 @@ fi
...
 
 git remote remove fork-to-test || echo "Added new remote fork-to-test"
-git remote add fork-to-test https://github.com/$SOURCE_GH_USER/$REPO_NAME.git
+git remote add fork-to-test git@github.com:$SOURCE_GH_USER/$REPO_NAME.git
```
Comment (jklukas) [inline git-push-fork-to-upstream-branch:21]: What effect does this change have?

Comment (jklukas) [reply git-push-fork-to-upstream-branch:21]: I tried this out locally, and it asked for me to enter my passphrase for my ssh key. I must have state saved locally for authenticating to GitHub via https, but not for ssh; I always clone using https. And I assume that you have local state set up for making ssh connections to GitHub. We might need to make this an option.

**Comment by jklukas** [comment]
See #3 which should provide support for the user to prefer ssh.

File: git-push-fork-to-upstream-branch (hunk 2 above)
Comment (harterrt) [reply git-push-fork-to-upstream-branch:21]: Bingo.

**Comment by harterrt** [comment]
Closing in favor of #3.

**Pull request closed without merging**